sdcm/nemesis.py

```diff
@@ -1082,13 +1074,28 @@ def add_ldap_configuration_to_node(node):
                 scylla_yaml.update(ldap_config)
             node.restart_scylla_server()
 
-        InfoEvent(message='Re-enable LDAP Authorization Configuration').publish()
-        create_ldap_container()
-        for node in self.cluster.nodes:
-            add_ldap_configuration_to_node(node)
+        try:
+            InfoEvent(message='Disable LDAP Authorization Configuration').publish()
+            for node in self.cluster.nodes:
+                remove_ldap_configuration_from_node(node)
+            destroy_ldap_container()
+
+            self.log.debug('Will wait few minutes with LDAP disabled, before re-enabling it')
+            time.sleep(600)
+
+            InfoEvent(message='Re-enable LDAP Authorization Configuration').publish()
+            create_ldap_container()
+            for node in self.cluster.nodes:
+                add_ldap_configuration_to_node(node)
+            if not ContainerManager.is_running(self.tester.localhost, 'ldap'):
+                raise LdapNotRunning("LDAP server was supposed to be running, but it is not")
+        finally:
+            if not ContainerManager.is_running(self.tester.localhost, 'ldap'):
+                InfoEvent(message='Re-enable LDAP Authorization Configuration').publish()
+                create_ldap_container()
```

Comment:
this is the part that we must be sure is working, AFAICT, maybe adding some more verification inside of it, to be sure we move "forward" only when it is correctly created, and a connection is functional.

Reply:
The problematic part turned out to be creating the connection to Ldap container.